File: collaboration-client/NimbleCollaborationClient/Type/CredentialType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.Script.Serialization;

namespace Nimble.Client.Type
{
    public class CredentialType : JsonType
    {

        public CredentialType(String username, String password) {
    	    this.username = username;
    	    this.password = password;
        }

        public String username { get; set; }
        public String password { get; set; }

	    public static CredentialType mapJson(String json) {
		    try {
                return new JavaScriptSerializer().Deserialize<CredentialType>(json);
            }
            catch (Exception e)
            {
			    return null;
		    }
	    }

    }
}

File: collaboration-service/src/main/java/it/domina/nimble/collaboration/auth/type/CredentialType.java
package it.domina.nimble.collaboration.auth.type;

import org.apache.log4j.Level;
import org.apache.log4j.Logger;

import com.google.gson.Gson;

import it.domina.nimble.utils.JsonType;

public class CredentialType extends JsonType {

	private static final Logger logger = Logger.getLogger(CredentialType	.class);

   	private String username;
   	private String password;
	
    public CredentialType(String username, String password) {
    	this.username = username;
    	this.password = password;
    }

    public String getUsername() {
    	return this.username;
    }

    public String getPassword() {
    	return this.password;
    }

	public static CredentialType mapJson(String json) {
		try {
            return (new Gson()).fromJson(json, CredentialType.class);
		} catch (Exception e) {
			logger.log(Level.INFO, e);
			return null;
		}
	}

}
